src/impls/cstring.rs

```diff
@@ -49,25 +66,51 @@ mod tests {
 
     use super::*;
 
-    #[rstest(input, expected, expected_rest,
+    #[rstest(input, len, expected, expected_rest,
         case(
             &[b't', b'e', b's', b't', b'\0'],
+            Some(5),
+            CString::new("test").unwrap(),
+            &[],
+        ),
+        case(
+            &[b't', b'e', b's', b't', b'\0'],
+            None,
             CString::new("test").unwrap(),
             &[],
         ),
         case(
             &[b't', b'e', b's', b't', b'\0', b'a'],
+            Some(5),
             CString::new("test").unwrap(),
             &[b'a'],
         ),
+        case(
+            &[b't', b'e', b's', b't', b'\0', b'a'],
+            None,
+            CString::new("test").unwrap(),
+            &[b'a'],
+        ),
+
+        #[should_panic(expected = "Incomplete(NeedSize { bits: 8 })")]
+        case(&[b't', b'e', b's', b't'], Some(5), CString::new("test").unwrap(), &[]),
 
         #[should_panic(expected = "Incomplete(NeedSize { bits: 8 })")]
-        case(&[b't', b'e', b's', b't'], CString::new("test").unwrap(), &[]),
+        case(&[b't', b'e', b's', b't'], None, CString::new("test").unwrap(), &[]),
```

Comment:
thought of a new test, what happens when no null byte is present

and, what happens when it's the correct len, but no nul byte present

Reply:
Both of these are represented by 141cc5b5ecd68ba63b9d7b97dbee06964ef2349e